lib/std/crypto/ghash.zig

```diff
@@ -144,38 +161,63 @@ pub const Ghash = struct {
             (z3 & 0x88888888888888888888888888888888) ^ extra;
     }
 
+    const I256 = struct {
+        hi: u128,
+        lo: u128,
+        mid: u128,
```

Comment:
I don't think I understand what `mid` is

Reply:
Schoolbook multiplication:

```text
ab * cd =

    bd
+  ad
+  bc
+ ac
```

`ac` is `hi`, `bd` is `lo` and in the `mid`dle we have `ad+bc`. Eventually, a shifted addition will get rid of it:

```text
mid = ad + bc
 lo = bd + (mid << n)
 hi = ac + (mid >> n)
```

We are doing several multiplications in a row. So, instead of adding the middle term after every multiplication, we can accumulate the `lo`, `hi` and `mid` values, and only add `mid` at the end. This is what we do here.